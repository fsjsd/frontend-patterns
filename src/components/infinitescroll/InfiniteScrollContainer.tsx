import React, { Suspense } from 'react'
import { ContentContainer } from '../../ux/ContentContainer'
import Loading from '../../ux/Loading'
import InfiniteScroll from './InfiniteScroll';
import { FeedItem, getItem } from './utils/data';
import { db } from "./utils/db";

// import RecyclingVirtualList from './recyclingvirtuallist/RecyclingVirtualList'
const RecyclingVirtualList = React.lazy(() => import('./recyclingvirtuallist/RecyclingVirtualList'))
//const InfiniteScroll = React.lazy(() => import('./InfiniteScroll'))

const DB = db(1000, 1000, getItem);

const InfiniteScrollContainer = () => {

  return (
    <ContentContainer
      title="Infinite Scroll"
      codeLink="/features/infinitescroll"
      markDownPromise={import('./requirements.md')}
    >
      <Suspense fallback={<Loading />}>
        <div role="main" style={{ width: 300, height: 500, border: "solid 1px red", overflowY: "scroll" }}>

          <InfiniteScroll />

          {false && <RecyclingVirtualList
            pageSize={10}
            load={(start, limit) => DB.load(start, limit).then((cursor) => cursor.chunk) as Promise<FeedItem[]>}
            itemFullComponent={(item: FeedItem) => <div>{item.name}</div>}
          />}

        </div>
      </Suspense>
    </ContentContainer>
  )
}

export default InfiniteScrollContainer
