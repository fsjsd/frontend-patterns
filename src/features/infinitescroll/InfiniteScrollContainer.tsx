import React, { Suspense } from 'react'
import ContentHeader, { ContentHeaderLabel } from '../../ux/ContentHeader'
import { ContentWrapper } from '../../ux/ContentWrapper'
import Loading from '../../ux/Loading'
//import ReactMarkdown from 'react-markdown'

const ReactMarkdown = React.lazy(() => import('react-markdown'))

const InfiniteScrollContainer = () => {
  return (
    <>
      <ContentHeader>
        <ContentHeaderLabel>Infinite Scroll</ContentHeaderLabel>
      </ContentHeader>
      <ContentWrapper>
        <Suspense fallback={<Loading />}>
          <ReactMarkdown># Hello, *world*!</ReactMarkdown>
        </Suspense>
      </ContentWrapper>
    </>
  )
}

export default InfiniteScrollContainer
