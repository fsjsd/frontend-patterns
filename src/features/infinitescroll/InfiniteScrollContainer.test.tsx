import React from 'react'
import { render } from '@testing-library/react'
import InfiniteScrollContainer from './InfiniteScrollContainer'

describe('InfiniteScrollContainer', () => {
  test('render matches snapshot', () => {
    const { container } = render(<InfiniteScrollContainer />)
    expect(container).toMatchSnapshot()
  })
})
