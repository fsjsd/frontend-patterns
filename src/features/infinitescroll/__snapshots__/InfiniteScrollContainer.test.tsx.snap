// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteScrollContainer render matches snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ dtMXGe"
  >
    <div
      class="sc-gsnTZi dkPlsI"
    >
      Infinite Scroll
    </div>
  </div>
  <div
    class="sc-dkzDqf eGEKCX"
  >
    <div
      class="sc-hKMtZM hbOElo"
    >
      <div />
      <div />
    </div>
  </div>
</div>
`;
